tests/functional_tests/website_load.bats

```diff
@@ -0,0 +1,156 @@
+  cmp "$BATS_TMPDIR/$TESTDIR/FileSaver.min.js" "$BATS_TMPDIR/$TESTDIR/js/FileSaver.min.js"
+  result=$?
+  [ "$result" -eq 0 ]
+
+  # check for socket.io.js
+  curl -s --ipv4 -k https://127.0.0.1:9000/socket.io/socket.io.js -o "$BATS_TMPDIR/$TESTDIR/socket.io.js"
+  cmp "$BATS_TMPDIR/$TESTDIR/socket.io.js" "$BATS_TMPDIR/$TESTDIR/js/socket.io.js"
+  result=$?
+  [ "$result" -eq 0 ]
+}
+
+@test "Load third-party css files" {
+  # check for css/*.css files
+  cp -f "data/$TESTDIR/js/package.json" "$BATS_TMPDIR/$TESTDIR/package.json"
+
+  # get a fresh copy of all the files to be tested from npm
+  npm --quiet install --prefix "$BATS_TMPDIR/$TESTDIR" 1>/dev/null
+
+  # copy only the necessary files to the required directories
+  mkdir -p "$BATS_TMPDIR/$TESTDIR/css"
+  cp -f "$BATS_TMPDIR/$TESTDIR/node_modules/materialize-css/dist/css/materialize.min.css" "$BATS_TMPDIR/$TESTDIR/css/"
+  rm -rf "$BATS_TMPDIR/$TESTDIR/node_modules"
+
+  # check for icon.css
+  curl -s --ipv4 -k https://127.0.0.1:9000/css/icon.css -o "$BATS_TMPDIR/$TESTDIR/icon.css"
+  cmp "$BATS_TMPDIR/$TESTDIR/icon.css" "data/$TESTDIR/css/icon.css"
+  result=$?
+  [ "$result" -eq 0 ]
+
+  # check for style.css
+  curl -s --ipv4 -k https://127.0.0.1:9000/css/style.css -o "$BATS_TMPDIR/$TESTDIR/style.css"
+  cmp "$BATS_TMPDIR/$TESTDIR/style.css" "data/$TESTDIR/css/style.css"
+  result=$?
+  [ "$result" -eq 0 ]
+
+  # check for materialize.min.css
+  curl -s --ipv4 -k https://127.0.0.1:9000/css/materialize.min.css -o "$BATS_TMPDIR/$TESTDIR/materialize.min.css"
+  cmp "$BATS_TMPDIR/$TESTDIR/materialize.min.css" "$BATS_TMPDIR/$TESTDIR/css/materialize.min.css"
+  result=$?
+  [ "$result" -eq 0 ]
+}
+
+# for the following tests, links are extracted from index.html
+# see http://stackoverflow.com/questions/9561020/how-do-i-use-the-python-scrapy-module-to-list-all-the-urls-from-my-website
+
+@test "Check all hyperlinks" {
+  # check for live status of all http(s) href URLs on the page.
+
+  # for the autograder home page
+  mkdir -p "$BATS_TMPDIR/hyperlinks"
+  curl -s --ipv4 -k https://127.0.0.1:9000 -o "$BATS_TMPDIR/$TESTDIR/index.html"
+  cmp "$BATS_TMPDIR/$TESTDIR/index.html" "data/$TESTDIR/index.html"
+  result=$?
+  [ "$result" -eq 0 ]
+
+  # for the repository home page on Github
+  curl -fsS --head --request GET https://github.com/AutolabJS/AutolabJS
+  result=$?
+  [ "$result" -eq 0 ]
+
+  # for the repository wiki page on Github
+  curl -fsS --head --request GET https://github.com/AutolabJS/AutolabJS/wiki
+  result=$?
+  [ "$result" -eq 0 ]
+
+  # for the making a submission page on Github
+  curl -fsS --head --request GET https://github.com/AutolabJS/AutolabJS/wiki/v0.2.0-Making-a-Submission
+  result=$?
+  [ "$result" -eq 0 ]
+
+  # for the post lab evaluation page on Github
+  curl -fsS --head --request GET https://github.com/AutolabJS/AutolabJS/wiki/v0.2.0-Post-lab-self-evaluation
```

Comment:
The latest page is `https://github.com/AutolabJS/AutolabJS/wiki/v0.3.0-Post-lab-self-evaluation`